opm/parser/eclipse/EclipseState/IOConfig/RestartConfig.cpp

```diff
@@ -360,6 +360,24 @@ RPTSCHED( const DeckKeyword& keyword ) {
 }
 
 
+    void RestartConfig::addRestartKeywords( size_t timeStep , const std::map<std::string, int>& input_mnemonics) {
```

Comment:
Ok, so I'm getting my nitpicking magnifying glass.

![nitpick](https://cloud.githubusercontent.com/assets/8667957/17847028/152bfe38-684c-11e6-9dfe-6cec62bd511e.jpg)

This approach can be made *far* more elegant by acknowledging that we're doing a copy anyway.

```
add( size_t ts, map< std::string, int > mnemonics ) {
    auto allprops = mnemonics.find( "ALLPROPS" );
    if( allprops == mnemonics.end() ) return this->restart_keywords.update( ts, mnemonics );
    auto value = allprops->second;
    mnemonics.erase( allprops );
    for( const auto& kw : { "BG", "BO", "..." } )
        mnemonics[ kw ] = value;
    this->restart_keywords.update( ts, mnemonics );
}
```